src/sentry/workflow_engine/processors/data_condition_group.py

```diff
@@ -4,13 +4,15 @@
 from sentry.utils.function_cache import cache_func_for_models
 from sentry.workflow_engine.models import DataCondition, DataConditionGroup
 from sentry.workflow_engine.processors.data_condition import split_conditions_by_speed
-from sentry.workflow_engine.types import DataConditionResult, ProcessedDataConditionResult
+from sentry.workflow_engine.types import DataConditionResult
 
 logger = logging.getLogger(__name__)
 
 T = TypeVar("T")
 
-DataConditionGroupResult = tuple[ProcessedDataConditionResult, list[DataCondition]]
+ProcessedDataConditionResult = tuple[bool, DataCondition, DataConditionResult]
+ProcessedDataConditionGroupResult = tuple[bool, list[tuple[DataCondition, DataConditionResult]]]
+DataConditionGroupResult = tuple[ProcessedDataConditionGroupResult, list[DataCondition]]
```

Comment:
would it be better to define these as dataclasses? giving a name to each item in the tuple might help a passerby understand what it's used for c: